Sources/PackageGraph/VersionSetSpecifier.swift

```diff
@@ -225,8 +276,8 @@ extension VersionSetSpecifier {
             }
 
             if lhs.lowerBound == rhs {
-                // Return empty if the range represents the exact version.
-                if lhs.lowerBound == lhs.upperBound {
+                // Return empty if the range represents the exact version or the range is empty.
+                if lhs.lowerBound == lhs.upperBound || lhs.lowerBound.nextPatch() == lhs.upperBound {
```

Comment:
I'm not sure I understand this part — could this not return true for a 1.0.1 <.. 1.0.2 type of situation where the next patch of 1.0.1 is 1.0.2?

Reply:
Yah, that's the intent, since 1.0.1..<1.0.2 only includes 1.0.1